cmd/bootkube/render.go

```diff
@@ -63,6 +71,7 @@ func init() {
 	cmdRender.Flags().BoolVar(&renderOpts.selfHostKubelet, "experimental-self-hosted-kubelet", false, "(Experimental) Create a self-hosted kubelet daemonset.")
 	cmdRender.Flags().StringVar(&renderOpts.cloudProvider, "cloud-provider", "", "The provider for cloud services.  Empty string for no provider")
 	cmdRender.Flags().BoolVar(&renderOpts.selfHostedEtcd, "experimental-self-hosted-etcd", false, "(Experimental) Create self-hosted etcd assets.")
+	cmdRender.Flags().BoolVar(&renderOpts.etcdUseTLS, "etcd-use-tls", false, "If true, uses TLS for etcd. Implicitly true if --etcd-ca-path,--etcd-certificate-path,--etcd-private-key-path are set. If true but those flags are not set etcd TLS certificates will be generated. Not supported if --experimental-self-hosted-etcd=true.")
```

Comment:
@aaronlevy @dghubble the generation of etcd assets here is mainly intended as a convenience to help with testing. I was under the assumption that Tectonic would generate etcd certs upstream and simply pass them into bootkube. That's the main use production use case I would envision.

I can keep support for insecure etcd as well if it's a huge pain point for Tectonic, but I agree with Aaron that it feels like a stopgap we should try to discourage as much as possible.